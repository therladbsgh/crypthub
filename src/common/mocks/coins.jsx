export const coins = [
    {
        name: 'Bitcoin',
        symbol: 'BTC',
        currPrice: 10000
    },
    {
        name: 'Ethereum',
        symbol: 'ETH',
        currPrice: 5000
    }
]
