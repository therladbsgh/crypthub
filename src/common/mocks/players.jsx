import date from 'date-and-time';

export const player1 = {
    name: 'player1',
    portfolio: [
        {
            name: 'Bitcoin',
            amount: 250
        },
        {
            name: 'Ethereum',
            amount: 1000.28373623
        },
        {
            name: 'USD',
            amount: 250
        }
    ],
    transactions: {
        history: [
            {
                id: '1',
                type: 'market',
                side: 'buy',
                size: 25,
                price: 100,
                symbol: 'BTC',
                date: date.parse('Apr 1 2018', 'MMM D YYYY'),
                GTC: true,
                expiration: null,
                filled: true,
                filledDate: date.parse('Apr 3 2018', 'MMM D YYYY')
            },
            {
                id: '2',
                type: 'limit',
                side: 'buy',
                size: 35.000006,
                price: 200,
                symbol: 'ETH',
                date: date.parse('Apr 2 2018', 'MMM D YYYY'),
                GTC: false,
                expiration: date.parse('Apr 5 2018', 'MMM D YYYY'),
                filled: false,
                filledDate: null
            }
        ],
        current: [
            {
                id: '3',
                type: 'short',
                side: 'sell',
                size: 80,
                price: 100,
                symbol: 'BTC',
                date: date.parse('Apr 2 2018', 'MMM D YYYY'),
                GTC: true,
                expiration: null,
                filled: false,
                filledDate: null
            },
            {
                id: '4',
                type: 'limit',
                side: 'buy',
                size: 2,
                price: 200,
                symbol: 'ETH',
                date: date.parse('Apr 7 2018', 'MMM D YYYY'),
                GTC: false,
                expiration: date.parse('May 2 2018', 'MMM D YYYY'),
                filled: false,
                filledDate: null
            }
        ]
    },
    netWorth: 200,
    numTrades: 80,
    netReturn: -20,
    todayReturn: 5,
    currRank: 2,
    buyingPower: 150,
    shortReserve: 50
};

export const player2 = {
    name: 'player2',
    portfolio: [
        {
            name: 'Bitcoin',
            amount: 300
        }
    ],
    transactions: {
        history: [
            {
                id: '5',
                side: 'buy',
                size: 1,
                price: 10,
                symbol: 'ETH'
            },
            {
                id: '6',
                side: 'sell',
                size: 3500,
                price: 202220,
                symbol: 'ETH'
            }
        ],
        current: [
            {
                id: '7',
                side: 'sell',
                size: 23230,
                price: 232,
                symbol: 'BTC'
            },
            {
                id: '8',
                side: 'buy',
                size: 6000,
                price: 50,
                symbol: 'ETH'
            }
        ]
    },
    netWorth: 300,
    numTrades: 2,
    netReturn: 120
};
