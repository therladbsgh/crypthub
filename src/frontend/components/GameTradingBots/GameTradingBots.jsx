import * as _ from 'lodash';
import React, { Component } from 'react';
import { Header, Dropdown, Form, Button, Message, TextArea } from 'semantic-ui-react';
import { GameBackend } from 'endpoints';
import { GameTradingBotsStyle as styles } from 'styles';

export default class GameTradingBots extends Component {
    constructor(props) {
        super(props);

        this.state = {
            setBotObj: {
                gameId: this.props.gameId,
                playerId: this.props.player._id,
                botId: ''
            },
            loading: false,
            err: ''
        };

        this.handleChange = this.handleChange.bind(this);
        this.handleSave = this.handleSave.bind(this);
        this.handleStop = this.handleStop.bind(this);
    }

    handleChange(event, { value }) {
		this.setState({
            setBotObj: {
                ...this.state.setBotObj,
                botId: value
            }
        });
    }

    handleSave() {
        const { setBotObj } = this.state;
        this.setState({
            loading: true,
            err: ''
        });

        GameBackend.setTradingBot(setBotObj)
		.then(res => {
            console.log('success! ', res);
            this.setState({
                setBotObj: {
                    ...setBotObj,
                    botId: ''
                },
                loading: false
            });
		}, ({ err }) => {
			console.log('error! ', err);
			this.setState({ loading: false, err });
        });
    }

    handleStop() {
        const { setBotObj } = this.state;

        this.setState({
            loading: true,
            err: ''
        });

        GameBackend.setTradingBot(_.set(_.clone(setBotObj), 'botId', ''))
		.then(res => {
            console.log('success! ', res);
            this.setState({ loading: false });
		}, ({ err }) => {
			console.log('error! ', err);
			this.setState({ loading: false, err });
        });
    }

    render() {
        const { player } = this.props;
        const { setBotObj, loading, err } = this.state;
        const { tradingBots, activeBotId } = player;
        const { botId } = setBotObj;

        return (
			<div>
				<Header as='h2'>Trading Bot Settings</Header>
                <Form loading={loading} error={!!err}>
                    <Form.Group>
                        <Form.Field width={5}>
                            <label>Trading Bot Currently in Play</label>
                            { activeBotId ? _.find(tradingBots, { _id: activeBotId }).name : 'You currently don\'t have an active trading bot.'}
                            <br /><br /><br />
                            <Button icon='stop' negative disabled={!activeBotId} onClick={this.handleStop} content='Stop Active Bot' />
                        </Form.Field>
                        <Form.Field width={5}>
                            <label>Trading Bot to Make Active</label>
                            <Dropdown placeholder='Trading bot to make active' search selection options={_.map(tradingBots, t => ({ text: t.name, value: t._id }))} value={botId} onChange={this.handleChange} />
                            <br /><br />
                            <Button icon='save' positive onClick={this.handleSave} content='Save Changes' />
                        </Form.Field>
                    </Form.Group>
                    <Form.Field width={10}>
                        <Message
                            error
                            header='Error'
                            content={err}
                        />
                    </Form.Field>
                </Form>
                <Header as='h2'>Debug Log</Header>
                <Form>
                    <Form.Field>
                        <TextArea className={styles.debugLog} name='debugLog' value={'debug log here'} />
                    </Form.Field>
                </Form>
			</div>
        );
    }
}
