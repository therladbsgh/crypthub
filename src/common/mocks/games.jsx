import date from 'date-and-time';
import * as PlayerMocks from './players';

export const game1 = {
    id: '1',
    name: 'game1',
    description: 'desc1',
    host: 'user1',
    created: date.parse('Apr 1 2018', 'MMM D YYYY'),
    start: date.parse('Apr 2 2018', 'MMM D YYYY'),
    end: date.parse('May 7 2018', 'MMM D YYYY'),
    playerPortfolioPublic: true,
    startingBalance: 10000,
    commissionValue: 10,
    shortSelling: true,
    limitOrders: true,
    stopOrders: true,
    lastUpdated: new Date(),
    completed: false,
    players: [
        PlayerMocks.player1,
        PlayerMocks.player2
    ],
    isPrivate: false,
    password: ''
};

export const game2 = {
    id: '2',
    name: 'game2',
    description: 'desc2',
    host: 'user2',
    created: date.parse('Apr 2 2018', 'MMM D YYYY'),
    start: date.parse('Apr 3 2018', 'MMM D YYYY'),
    end: date.parse('May 1 2018', 'MMM D YYYY'),
    playerPortfolioPublic: true,
    startingBalance: 10000,
    commissionValue: 10,
    shortSelling: true,
    limitOrders: true,
    stopOrders: true,
    lastUpdated: new Date(),
    completed: false,
    players: [
        PlayerMocks.player1,
        PlayerMocks.player2
    ],
    isPrivate: false,
    password: ''
};

export const game3 = {
    id: '3',
    name: 'game3',
    description: 'desc3',
    host: 'user3',
    created: date.parse('Apr 4 2018', 'MMM D YYYY'),
    start: date.parse('Apr 8 2018', 'MMM D YYYY'),
    end: date.parse('Apr 28 2018', 'MMM D YYYY'),
    playerPortfolioPublic: true,
    startingBalance: 10000,
    commissionValue: 10,
    shortSelling: true,
    limitOrders: true,
    stopOrders: true,
    lastUpdated: new Date(),
    completed: false,
    players: [
        PlayerMocks.player1,
        PlayerMocks.player2
    ],
    isPrivate: false,
    password: ''
};
