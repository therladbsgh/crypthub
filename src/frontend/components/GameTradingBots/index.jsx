import GameTradingBots from './GameTradingBots';

export default GameTradingBots;
