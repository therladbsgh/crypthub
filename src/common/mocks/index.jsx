import * as GameMocks from './games';
import * as PlayerMocks from './players';

export { GameMocks, PlayerMocks };
