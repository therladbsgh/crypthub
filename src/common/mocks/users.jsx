import * as GameMocks from './games';

export const user1 = {
    id: 'id1',
    name: 'user1',
    ELO: 3000,
    games: [
        GameMocks.game1,
        GameMocks.game2
    ]
};

export const user2 = {
    id: 'id2',
    name: 'user2',
    ELO: 2000,
    games: [
        GameMocks.game2,
        GameMocks.game3
    ]
};

export const user3 = {
    id: 'id3',
    name: 'user3',
    ELO: 2000,
    games: [
        GameMocks.game2,
        GameMocks.game3
    ]
};
