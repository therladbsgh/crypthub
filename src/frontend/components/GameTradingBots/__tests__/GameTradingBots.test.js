import React from 'react';
import GameTradingBots from '../GameTradingBots';
import { GameMocks } from 'mocks';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  	const tree = renderer.create(<GameTradingBots player={GameMocks.game1.players[0]} />).toJSON();
  	expect(tree).toMatchSnapshot();
});