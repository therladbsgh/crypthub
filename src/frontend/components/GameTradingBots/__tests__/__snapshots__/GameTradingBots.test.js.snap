// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h2
    className="ui header"
  >
    Trading Bot Settings
  </h2>
  <form
    action={undefined}
    className="ui form"
    onSubmit={[Function]}
  >
    <div
      className="fields"
    >
      <div
        className="five wide field"
      >
        <label>
          Trading Bot Currently in Play
        </label>
        tradingbot1
        <br />
        <br />
        <br />
        <button
          className="ui negative button"
          disabled={undefined}
          onClick={[Function]}
          role="button"
          tabIndex={undefined}
        >
          <i
            aria-hidden="true"
            className="stop icon"
          />
          Stop Active Bot
        </button>
      </div>
      <div
        className="five wide field"
      >
        <label>
          Trading Bot to Make Active
        </label>
        <div
          aria-busy={undefined}
          aria-disabled={undefined}
          aria-expanded={false}
          className="ui search selection dropdown"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          role="combobox"
          tabIndex={undefined}
        >
          <input
            aria-autocomplete="list"
            autoComplete="off"
            className="search"
            onChange={[Function]}
            style={
              Object {
                "width": undefined,
              }
            }
            tabIndex={0}
            type="text"
            value=""
          />
          <div
            aria-live="polite"
            className="default text"
            role="alert"
          >
            Trading bot to make active
          </div>
          <i
            aria-hidden="true"
            className="dropdown icon"
            onClick={[Function]}
          />
          <div
            aria-multiselectable={undefined}
            className="menu transition"
            role="listbox"
          >
            <div
              aria-checked={false}
              aria-disabled={undefined}
              aria-selected={true}
              className="selected item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <span
                className="text"
              >
                tradingbot1
              </span>
            </div>
            <div
              aria-checked={false}
              aria-disabled={undefined}
              aria-selected={false}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <span
                className="text"
              >
                tradingbot2
              </span>
            </div>
          </div>
        </div>
        <br />
        <br />
        <button
          className="ui positive button"
          disabled={undefined}
          onClick={[Function]}
          role="button"
          tabIndex={undefined}
        >
          <i
            aria-hidden="true"
            className="save icon"
          />
          Save Changes
        </button>
      </div>
    </div>
    <div
      className="ten wide field"
    >
      <div
        className="ui error message"
      >
        <div
          className="content"
        >
          <div
            className="header"
          >
            Error
          </div>
          <p>
            
          </p>
        </div>
      </div>
    </div>
  </form>
  <h2
    className="ui header"
  >
    Debug Log
  </h2>
  <form
    action={undefined}
    className="ui form"
    onSubmit={[Function]}
  >
    <div
      className="field"
    >
      <textarea
        className={undefined}
        name="debugLog"
        onChange={[Function]}
        onInput={[Function]}
        rows={3}
        style={
          Object {
            "resize": "",
          }
        }
        value="debug log here"
      />
    </div>
  </form>
</div>
`;
